actionsystemtest/lib/action_system_test/server.rb

```diff
@@ -0,0 +1,34 @@
+require "rack/handler/puma"
+
+module ActionSystemTest
+  class Server # :nodoc:
+    def initialize(port)
+      @port = port
+    end
+
+    def run
+      register
+      setup
+    end
+
+    private
+      def register
+        Capybara.register_server :puma do |app, host|
```

Comment:
How about registering this as :rails_puma rather than :puma so as not to override Capybaras default :puma server registration.  Additionally, Capybara calls registered server blocks with |app, port, host| - not |app,host|